hyperspy/model.py

```diff
@@ -1480,7 +1483,11 @@ def print_current_values(self, only_free=True):
                 parameters = component.free_parameters if only_free \
                     else component.parameters
                 for parameter in parameters:
-                    if not hasattr(parameter.value, '__iter__'):
+                    if hasattr(parameter.value, '__iter__'):
+                        if skip_multi:
+                            for idx in range(len(parameter.value)):
+                                print("\t\ta[%d]\t%g" % (idx, parameter.value[idx]))
+                    else:
```

Comment:
Yes, of course, it should be `if not skip_multi`.

Otherwise I am a bit at a loss. I only realized now that the `__iter__`able parameters can be more complex, like dictionaries as in the EELS model. Is there a generic way to iterate through the parameters that catch tuples, arrays, ..., dictionaries? Do I have to deal with things like dictionaries of dictionaries? Or a list that contains a dictionary and a list?